source/plot/ScatterPlotData2D.ooc

```diff
@@ -57,12 +63,36 @@ ScatterPlotData2D: class extends PlotData2D {
 		halfLineHeight := (fontSize as Float) / 2.0f
 		match (this shape) {
 			case Shape Circle =>
-				result = result & Shapes circle(FloatPoint2D new(start x + halfLineHeight, start y), size / 2.0f, this opacity, this color)
+				result = result & Shapes circle(FloatPoint2D new(start x + halfLineHeight, start y), size / 2.0f, this colorBgra)
 			case Shape Square =>
-				result = result & Shapes rect(FloatPoint2D new(start x, start y - halfLineHeight), FloatPoint2D new(size, size), this opacity, this color)
+				result = result & Shapes rect(FloatPoint2D new(start x, start y - halfLineHeight), FloatPoint2D new(size, size), this colorBgra)
 			case =>
 		}
-		result = result & Shapes text(FloatPoint2D new(start x + fontSize as Float, start y + halfLineHeight), this label, fontSize, this opacity, this color)
+		result = result & Shapes text(FloatPoint2D new(start x + fontSize as Float, start y + halfLineHeight), this label, fontSize, this colorBgra)
+		result
+	}
+
+	minValues: override func -> FloatPoint2D {
+		result: FloatPoint2D
+		if (this dataSeries empty)
+			result = FloatPoint2D new()
+		else {
+			result = this dataSeries[0]
+			for (i in 1 .. this dataSeries count)
+				result = result minimum(this dataSeries[i])
+		}
+		result
+	}
```

Comment:
Haha, oops, no, it isn't! But it should be, so move this function there, at least the part inside the `else` clause. I'm not sure what should be returned in case of an empty list...